### DISCUSSION TITLE
django-channels

### DISCUSSION THREAD

**Discussion by mronoffon** [django-tenants/django-tenants#645] [Q&A]
hello

anyone got django-channels working with django-tenants (also using gunicorn/uvicorn)?  if so, would you mind
sharing your asgi.py and any middleware needed?

thanks in advance

**Comment by tomturner** [comment]
Yes you can use django-channels. However you can’t use gunicorn with
channels I suggest you use daphne as per the channel docs suggest.

**Comment by mronoffon** [comment]
do you know of any examples that use the PUBLIC_SCHEMA_URLCONF, ROOT_URLCONF and ASGI_APPLICATION together?

i have daphne installed to handle sockets and gunicorn to handle web traffic, but keep running into errors with the ProtocolTypeRouter/ URLRouter and djang-tenant urls

thanks in advance
 
> Yes you can use django-channels. However you can’t use gunicorn with
> channels I suggest you use daphne as per the channel docs suggest.
>
